src/store/staking/staking.sagas.ts

```diff
@@ -105,13 +112,68 @@ export function* fetchStakesList() {
   }
 }
 
+export function* fetchHistoryStakeEvents() {
+  const subgraphClient = yield* select(subgraphClientSelector);
+
+  if (!subgraphClient) throw new Error('Wallet not connected');
+
+  const { stakeEvents } = yield* call(historyStakesQuery, subgraphClient);
+
+  return stakeEvents;
+}
+
+export function* filterContractStakedEvents(
+  stakeEvents: StakeHistoryQueryItem[]
+) {
+  const account = yield* select(accountSelector);
+  const vestsList = yield* select(vestsListSelector);
+
+  if (!account || !vestsList) {
+    throw new Error('Wallet not connected');
+  }
+
+  return stakeEvents.filter(
+    ({ staker }) =>
+      staker === account.toLowerCase() ||
```

Comment:
I think way more efficient way would be to filter the stakes inside the query using `where`.  There will probably thousand of  stakes from all users in the system. I don't think fetching them all and filtering on the front end side is a good idea